addons/mail/static/src/new/core/channel_member_model.js

```diff
@@ -5,10 +5,25 @@
  * @typedef Data
  * @property {number} id
  * @property {number} partnerId
+ * @property {import("@mail/new/core/thread_model").Thread} memberThread
  */
 export class ChannelMember {
-    constructor(_state, { id, partnerId }) {
-        Object.assign(this, { _state, id, partnerId });
+    static insert(state, data) {
+        let channelMember = state.channelMembers[data.id];
+        if (!channelMember) {
+            channelMember = new ChannelMember();
+            state.channelMembers[data.id] = channelMember;
+        }
+        Object.assign(channelMember, {
+            id: data.id,
+            partnerId: data.partnerId,
+            memberThread: data.memberThread,
```

Comment:
Note that while doing this, not providing a key in data would mean writing undefined on the record for that key, what should happen instead is to keep the current value.